### DISCUSSION TITLE
can we load options dynamically ?? Async way to load child options to parent ??

### DISCUSSION THREAD

**Discussion by Ilankumaran-developer** [dowjones/react-dropdown-tree-select#646] [Q&A]
I am in a project where the data set for the dropdown-tree is massive, Is there any provision here so that i can add children only when their parent is clicked and fetching children of that parent ??

**Comment by gandhis1** [comment]
I have the same issue, for what it's worth. When you have hundreds of thousands or millions of nodes, it's not fast enough.

As far as I know, this component is fully client-side and so to have server-side rendering and lazy loading you'd need to roll your own. It sounds like a very complex feature, so not getting my hopes up.

**Comment by mrchief** [comment]
This component can handle extremely large nodes - you can check the demo where it loads 15K plus nodes with ease. It does that by windowing the nodes, there is nothing to enable, it's just standard behavior. Would be curious to see how it's not working for you, if you have a code sandbox then please share.
